cmd/promtool/main.go

```diff
@@ -103,6 +104,20 @@ func main() {
 		"The unit test file.",
 	).Required().ExistingFiles()
 
+	backfillCmd := app.Command("backfill", "Backfill Prometheus data.")
+	backfillRuleCmd := backfillCmd.Command("rules", "Backfill Prometheus data for new rules.")
+	backfillRuleURL := backfillRuleCmd.Flag("url", "Prometheus API url.").Required().String()
+	backfillRuleEvalInterval := backfillRuleCmd.Flag("evaluation_interval", "How frequently to evaluate rules when backfilling.").
+		Default("1m").Duration()
+	backfillRuleStart := backfillRuleCmd.Flag("start", "If a start time is provided, the new rule backfilling will begin at this time. The default evaluates from the first know data. Start time should be RFC3339 or Unix timestamp.").
```

Comment:
You can't know when the first known data is from query_range alone without searching, in practice this will have to be provided by the user.

know->known